### DISCUSSION TITLE
Publii as a Wordpress replacement - how ready is it?

### DISCUSSION THREAD

**Discussion by IO101010** [GetPublii/Publii#1891] [General]
I'm just wondering how people find Publii as replacement for something like Wordpress. I really like it so far and we got so tired of maintaining Wordpress when plugins update. There's also the security concern. This is all too much for a simple static website like ours. We moved to GitLab Pages which has been much better for me as the maintainer, but it has made things more difficult for editor. I'm hoping Publii can help here.

I'd like to recreate our website in Publii, set it up to push new changes to the server and then hand it over to the site editor - who is not a developer. 

I see a post here about issues with Gitlab from quite a while back with no response: https://github.com/GetPublii/Publii/issues/1693

I know 0.x means don't use this for something mission critical, but would users and devs consider it stable and actively maintained?

**Comment by michaelfluegge** [comment]
From my experience it works like a charm, as long as you use it the way the developer designed it.
some people experiencing issues when tweaking Publii, but from my point of view the problems come from ignoring/not reading the rules described in the user and developer docs. 

---
For example, I built a site with Publii with the design of a WordPress template (not all of the functionality), and it works pretty well. I copied all the HTML/CSS and filled it with the handlebar stuff to make it dynamic. 

But I do not have security experience as I do not use a forum or anything else like that.

**Reply by IO101010** [reply]
Thanks, that's very helpful. Maybe Publii is more suitable than I thought. Are there any limitations on custom HTML / CSS? I would need to recreate the homepage exactly as it exists on the current site ( https://optimumhealth.ie ). I've added the code for the header and footer in the "Head" and "Footer" sections under Custom HTML. But when I add custom HTML to my homepage (minus the header and fuoter), the layout for the body is wrong. (Header and footer look fine, though)

**Comment by michaelfluegge** [comment]
You should read about the theme structure (https://getpublii.com/dev/theme-structure/) and go for building your theme from scratch, I think. 
As an example, I took the Wordpress theme Wilson (https://wilson.andersnoren.se) and rebuilt it using Publii (https://fluegge.net/ohnein/). I learned how handlebars work etc by reading the documentation (https://getpublii.com/dev/introduction-tags/) etc, and asking the forum, of course. It took some time but was worth the effort to get rid of Wordpress.

**Reply by IO101010** [reply]
Thank you or the links. I'll check it out. That site is actually hosted on Gitlab, but was exported from Wordpress.

Final question for now, if you can answer: How does technical collaboration work with Publii? With Wordpress, I might get an email from an editor saying that the layout of their post or page is problematic and I'll be able to log in and make changes. If I set up Publii on their computer and they need help - I imagine logging into their repo will break the workflow since their local Publii app won't know anything about it. I'd like them to be able to get support when they need it, including when I'm not around to help.

I see upwork.com has Publii devs there which is a good start, but I wonder how my client would get 3rd party help with Publii when they need it.

**Reply by michaelfluegge** [reply]
@bobmitro , I think, you should answer this.

**Comment by treeandcoffee** [comment]
I don't want to go into the workflow and third party support, but the question of whether Publii can serve as a Wordpress replacement.

Yes, I think that many of Publii's features and capabilities are more than sufficient for many SME clients' websites. Publii also offers important functions (webp conversion, SEO, Menus) that are only possible with WP as plugins or special themes. It seems important that, as @michaelfluegge mentioned, Publii is used as it is intended and not as we know managing a website from WP or other CMS. 
For webdevs, however, this also means that you have to develop or rethink new ways and possibilities when it comes to working colloboratively with Publii, for example Dropbox or other prospective new ways in the future. 

Personally, I like having an application that generates the website for me. It helps me to focus more on the deliverable content instead of config or settings.

**Reply by IO101010** [reply]
I have to agree that it's a decent Wordpress replacement. I've been looking through the showcase of commercial sites using it and they're impressive. Services like Dropbox are an interesting idea for collaboration. Thanks for mentioning that, I think that's my main concern so if Publii is portable enough to be used that way then I can commit a bit more to it. I think the outstanding issue for me will be reproducing our homepage layout. Ourselves and our customers like it as is.
